src/Workspaces/Core/Portable/CodeFixes/ExportCodeFixProviderAttribute.cs

```diff
@@ -26,6 +28,18 @@ public sealed class ExportCodeFixProviderAttribute : ExportAttribute
         /// </summary>
         public string[] Languages { get; }
 
+        /// <summary>
+        /// The document kinds for which this provider can provide code fixes. See <see cref="TextDocumentKind"/>.
+        /// By default, the provider supports code fixes only for source documents, <see cref="TextDocumentKind.Document"/>.
+        /// </summary>
+        public TextDocumentKind[] DocumentKinds { get; set; }
+
+        /// <summary>
+        /// The document extensions for which this provider can provide code fixes.
+        /// By default, this value is null and the document extension is not considered to determine applicability of code fixes.
```

Comment:
Same comment as elsewhere: we should document if this includes a leading period or not.